modules/ibc/handler.go

```diff
@@ -0,0 +1,199 @@
+package ibc
+
+import (
+	"fmt"
+
+	"github.com/tendermint/go-wire/data"
+	"github.com/tendermint/tmlibs/log"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/stack"
+	"github.com/tendermint/basecoin/state"
+)
+
+const (
+	// NameIBC is the name of this module
+	NameIBC = "ibc"
+	// OptionRegistrar is the option name to set the actor
+	// to handle ibc chain registration
+	OptionRegistrar = "registrar"
+)
+
+var (
+	allowIBC = []byte{0x42, 0xbe, 0xef, 0x1}
+)
+
+// AllowIBC is the special code that an app must set to
+// enable sending IBC packets for this app-type
+func AllowIBC(app string) basecoin.Actor {
+	return basecoin.Actor{App: app, Address: allowIBC}
+}
+
+// Handler allows us to update the chain state or create a packet
```

Comment:
lets try to avoid first person/s. `Handler allows the chain state to be updated or a packet to be created`